app/ts-cli/geminicli/export_test.go

```diff
@@ -0,0 +1,862 @@
+			}
+
+			if i%35 == 0 {
+				delete(fields, "field3_bool")
+			}
+		}
+
+		r := influx.Row{}
+
+		// fields init
+		r.Fields = make([]influx.Field, len(fields))
+		j := 0
+		for k, v := range fields {
+			r.Fields[j].Key = k
+			switch v.(type) {
+			case int64:
+				r.Fields[j].Type = influx.Field_Type_Int
+				r.Fields[j].NumValue = float64(v.(int64))
+			case float64:
+				r.Fields[j].Type = influx.Field_Type_Float
+				r.Fields[j].NumValue = v.(float64)
+			case string:
+				r.Fields[j].Type = influx.Field_Type_String
+				r.Fields[j].StrValue = v.(string)
+			case bool:
+				r.Fields[j].Type = influx.Field_Type_Boolean
+				if v.(bool) == false {
+					r.Fields[j].NumValue = 0
+				} else {
+					r.Fields[j].NumValue = 1
+				}
+			}
+			j++
+		}
+
+		sort.Sort(r.Fields)
+
+		vInt++
+		vFloat += 1.1
+		tags := map[string]string{
+			"tagkey1":    fmt.Sprintf("tagvalue1_%d", tv1),
+			`tag ke,y=2`: fmt.Sprintf("tagvalue2_%d", tv2),
+			"tagkey3":    fmt.Sprintf("tagvalue3_%d", tv3),
+			"tagkey4":    fmt.Sprintf(`ta valu,e=4_%d`, tv4),
+		}
+
+		// tags init
+		r.Tags = make(influx.PointTags, len(tags))
+		j = 0
+		for k, v := range tags {
+			r.Tags[j].Key = k
+			r.Tags[j].Value = v
+			j++
+		}
+		sort.Sort(&r.Tags)
+		tv4++
+		tv3++
+		tv2++
+		tv1++
+
+		name := mms(i)
+		r.Name = name
+
+		offset := time.Duration(random.Int63n(int64(24 * time.Hour)))
+		tm := basicTime.Add(offset)
+		r.Timestamp = tm.UnixNano()
+		r.UnmarshalIndexKeys(indexKeyPool)
+		r.UnmarshalShardKeyByTag(nil)
+		tm = tm.Add(interval)
+
+		pts = append(pts, r)
+	}
+	if pointNumOfPerSeries > 1 {
+		copyRs := copyPointRows(pts, pointNumOfPerSeries-1, interval, inc)
+		pts = append(pts, copyRs...)
+	}
+
+	sort.Slice(pts, func(i, j int) bool {
+		return pts[i].Timestamp < pts[j].Timestamp
+	})
+
+	return pts, pts[0].Timestamp, pts[len(pts)-1].Timestamp
+}
+func copyPointRows(rs []influx.Row, copyCnt int, interval time.Duration, inc bool) []influx.Row {
+	copyRs := make([]influx.Row, 0, copyCnt*len(rs))
+	for i := 0; i < copyCnt; i++ {
+		cnt := int64(i + 1)
+		for index, endIndex := 0, len(rs); index < endIndex; index++ {
+			tmpPointRow := rs[index]
+			tmpPointRow.Timestamp += int64(interval) * cnt
+			// fields need regenerate
+			if inc {
+				tmpPointRow.Fields = make([]influx.Field, len(rs[index].Fields))
+				if len(rs[index].Fields) > 0 {
+					tmpPointRow.Copy(&rs[index])
+					for idx, field := range rs[index].Fields {
+						switch field.Type {
+						case influx.Field_Type_Int:
+							tmpPointRow.Fields[idx].NumValue += float64(cnt)
+						case influx.Field_Type_Float:
+							tmpPointRow.Fields[idx].NumValue += float64(cnt)
+						case influx.Field_Type_Boolean:
+							tmpPointRow.Fields[idx].NumValue = float64(cnt & 1)
+						case influx.Field_Type_String:
+							tmpPointRow.Fields[idx].StrValue += fmt.Sprintf("-%d", cnt)
+						}
+					}
+				}
+			}
+
+			copyRs = append(copyRs, tmpPointRow)
+		}
+	}
+	return copyRs
+}
+
+func removeTestDirs() {
```

Comment:
If `t.TempDir()` is used, the testing package will automatically remove the temporary directory when the test case finishes.

Reply:
refactored